src/StateL1APCu.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+/**
+ * @file
+ * L1 state manager implementation.
+ */
+
+namespace LCache;
+
+/**
+ * L1 statistics manager storing in APCu.
+ *
+ * @author ndobromirov
+ */
+class StateL1APCu implements StateL1Interface
+{
+    /** @var string */
+    private $pool;
+
+    /** @var string */
+    private $statusKeyHits;
+
+    /** @var string */
+    private $statusKeyMisses;
+
+    /** @var string */
+    private $statusKeyLastAppliedEventId;
+
+    /**
+     * Constructor implementation.
+     *
+     * @param string $pool
+     *   Pool string that this state manager will collect statistics for.
+     */
+    public function __construct($pool)
+    {
+        $this->pool = $pool;
+
+        // Using designated variables to speed up key generation during runtime.
+        $this->statusKeyHits = 'lcache_status:' . $this->pool . ':hits';
+        $this->statusKeyMisses = 'lcache_status:' . $this->pool . ':misses';
+        $this->statusKeyLastAppliedEventId = 'lcache_status:' . $this->pool . ':last_applied_event_id';
+    }
+
+    /**
+     * {inheritdoc}
+     */
+    public function recordHit()
+    {
+        return $this->_recordEvent($this->statusKeyHits);
+    }
+
+    /**
+     * {inheritdoc}
+     */
+    public function recordMiss()
+    {
+        return $this->_recordEvent($this->statusKeyMisses);
+    }
+
+    /**
+     * Utility method to reduce code duplication.
+     *
+     * @param string $key
+     *   Key to store the evet counters in.
+     *
+     * @return bool
+     *   True on success, false otherwise.
+     */
+    private function _recordEvent($key)
```

Comment:
It's a personal preference. I like to have all private stuff prefixed with underscore old habit left from development on [Yii framework](http://www.yiiframework.com/doc/guide/1.1/en/basics.convention#code). If it's not ok with the project's naming conventions it fairly easy to change. Though the code sniffer did not complained for that :).